src/parse_input.py

```diff
@@ -0,0 +1,153 @@
+import os
+import sys
+import subprocess
+import json
+import xml.etree.ElementTree as ET
+import cyclus
+
+
+def parse_input(input, commodity_dictionary):
+    '''
+    archetype_commods = parse_input(input, commodity_dictionary)
+
+    inputs:
+        - input: a Cyclus XML input file
+        - commodity_dictionary: a dictionary with the Cyclus archetypes
+        available and the archetype tags of their incommodities and
+        outcommodities
+
+    outputs:
+        - facility_dictionary: a dictionary of each facility and its
+        incommodities and outcommodities. format:
+        {'facility' : (['incommodities'], ['outcommodities'])}
+    '''
+
+    tree = ET.parse(input)
+    root = tree.getroot()
+
+    archetypes_in_input = archetypes_in_input_file(root)
+    facility_dictionary = get_facility_and_commod_names(root,
+                                                        archetypes_in_input,
+                                                        commodity_dictionary)
+
+    return facility_dictionary
+
+
+def get_facility_and_commod_names(root, archetypes_in_input,
+                                  commodity_dictionary):
+    '''
+    facility_dict = get_facility_and_commod_names(root, archetypes_in_input,
+                                    commodity_dictionary)
+
+    get_facility_and_commodity_names reads a Cyclus input file element tree and
+    finds the facility names for each facility specified in a <facility> tag.
+    Then, using the facility archetype and module, the input and output
+    commodities are retrieved.
+
+    inputs:
+        - root: an xml.etree.ElementTree root of a Cyclus input file tree
+        - archetypes_in_input: a dictionary with format
+                               {'archetype' : 'module::archetype'}
+        - commodity_dictionary: a dictionary with the Cyclus archetypes
+        available and the archetype tags of their incommodities and
+        outcommodities
+
+    outputs:
+        - facility_dict: a dictionary of each facility and its
+        incommodities and outcommodities. format:
+        {'facility' : (['incommodities'], ['outcommodities'])}
+    '''
+
+    facility_dict = {}
+
+    for facility in root.findall('./facility'):
+        facility_name = facility.find('name').text
+        facility_archetype = facility.find('config/').tag
+        facility_module = archetypes_in_input[facility_archetype]
+        commodities = commodity_dictionary[facility_module]
+
+        in_commod_tags = commodities[0]
+        out_commod_tags = commodities[1]
+
+        facility_in_commods = []
+        facility_out_commods = []
+
+        for archetype_tag in facility.find('.config/' + facility_archetype):
+
+            in_commods = find_commod(archetype_tag, in_commod_tags)
```

Comment:
There is probably a way to setup a loop over in & out since it's the same code, but I'm trying to decide whether or not it would be too obfuscating.... we can discuss if you like